### PR TITLE
Plugin SDK Host Emulator

### PR DESCRIPTION
**Pull request by dmitry-yudakov**
The [original PR](https://github.com/momentum-xyz/ui-client/pull/369) got closed after Daniel merged and deleted his branch, so posting it again with the comments resolved - one component per file, comment with code snippet for HostEmulator, adjusted to latest PluginInterface.

### CODE REVIEW

File: packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        set: (field: string, value: unknown) => {
+          stateAttribute.current[field] = value;
+          return Promise.resolve();
+        }
+      }
+    }),
+    [theme]
+  );
+
+  const [topBar, setTopBar] = useState(<span />);
```
Comment (PAKovalev) [inline packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx:34]: Type of `useState`

File: packages/sdk/src/emulator/HostEmulator.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import {PluginInterface} from 'interfaces';
+
+import * as styled from './HostEmulator.styled';
+import {SpaceTabEmulator} from './components/';
+
+interface PropsInterface {
+  plugin: PluginInterface;
+}
+
+/**
```
Comment (PAKovalev) [inline packages/sdk/src/emulator/HostEmulator.tsx:11]: Removal?

File: packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  console.log('RENDER SpaceEmulator', {plugin});
+  const theme = useTheme();
+
+  const stateAttribute = useRef<Record<string, any>>({});
+
+  const coreProps: CorePluginPropsInterface = useMemo(
+    () => ({
+      theme: theme as ThemeInterface,
+      isSpaceAdmin: false,
+      spaceId: '123',
```
Comment (PAKovalev) [inline packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx:22]: `spaceId: '00000000-0000-0000-0000-000000000000'`

Let's use empty guid.

**Review by PAKovalev — APPROVED** [review]

**Review by Daboo1998 — APPROVED** [review]

File: packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx (hunk 1 above)
Comment (dmitry-yudakov) [reply packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx:34]: It's defined by the default value, why make it complex?

File: packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx (hunk 3 above)
Comment (dmitry-yudakov) [reply packages/sdk/src/emulator/components/SpaceTabEmulator/SpaceTabEmulator.tsx:22]: Yeah, good idea about guid format, but maybe not empty for less confusion.

File: packages/sdk/src/emulator/HostEmulator.tsx (hunk 2 above)
Comment (dmitry-yudakov) [reply packages/sdk/src/emulator/HostEmulator.tsx:11]: It's not a commented code but an example code snippet :) it should be this way

**Pull request merged**